lib/rspec/matchers/built_in/satisfy.rb

```diff
@@ -31,6 +33,28 @@ def failure_message
         def failure_message_when_negated
           "expected #{actual_formatted} not to #{description}"
         end
+
+      private # rubocop:disable Lint/UselessAccessModifier
+
+        if RSpec::Support::RubyFeatures.ripper_supported?
+          def block_representation
+            snippet = extract_block_snippet
+            snippet ? "expression #{snippet}" : 'block'
+          end
+
+          def extract_block_snippet
+            return nil unless @block
+            lines = Expectations::BlockSnippetExtractor.body_content_lines_of(@block, 'satisfy')
+            return nil unless lines.size == 1
+            "`#{lines.first}`"
+          rescue Expectations::BlockSnippetExtractor::Error
+            nil
+          end
```

Comment:
This has a lot of duplicated knowledge between this method and the similar method in the `change` matcher.  Is there a way to extract something common?  Or is that premature?